database_cleanup/model/purge_tables.py

```diff
@@ -110,9 +110,9 @@ def find(self, cr, uid, context=None):
             known_tables += [
                 column._sql_names(model_pool)[0]
                 for column in model_pool._columns.values()
-                if column._type == 'many2many'
+                if column._type == 'many2many' and
                 # unstored function fields of type m2m don't have _rel
-                and hasattr(column, '_rel')
+                hasattr(column, '_rel')
```

Comment:
What about:

    known_tables += [
        column._sql_names(model_pool)[0]
        for column in model_pool._columns.values()
        if (column._type == 'many2many' and
            # unstored function fields of type m2m don't have _rel
            hasattr(column, '_rel'))
    ]